ffi/ffi.cpp

```diff
@@ -4,6 +4,100 @@
 
 using namespace cvc5;
 
+// ## `Except Error α` constructors
+
+extern "C" lean_obj_res except_ok(
+  lean_obj_arg alpha,
+  lean_obj_arg val
+);
+
+extern "C" lean_obj_res except_ok_bool(
+  uint8_t val
+);
+
+extern "C" lean_obj_res except_ok_u32(
+  uint32_t val
+);
+
+extern "C" lean_obj_res except_ok_u8(
+  uint8_t val
+);
+
+extern "C" lean_obj_res except_err(
+  lean_obj_arg alpha,
+  lean_obj_arg msg
+); 
+
+// ## Exception-catching macro for `Except`
+//
+// Runs `code`, `return`s an `Except Error α` error on exceptions.
+
+#define CVC5_LEAN_API_TRY_CATCH_EXCEPT_BEGIN() \

```

Comment:
Let's only use parenthesis when it's needed! (i.e., `CVC5_LEAN_API_TRY_CATCH_EXCEPT_BEGIN`)